GUI.cpp

```diff
@@ -333,7 +333,8 @@ void draw_report_border(const ALLEGRO_FONT *font, float x, float y, int flags, c
 
 void draw_announcements(const ALLEGRO_FONT *font, float x, float y, int flags, std::vector<df::report *> &announcements)
 {
-    for (int i = announcements.size() - 1; i >= 0 && announcements[i]->duration > 0; i--)
+    int maxAnnouncements = 10;
+    for (int i = announcements.size() - 1; i >= (announcements.size() - maxAnnouncements) && announcements[i]->duration > 0; i--)
```

Comment:
will crash if there are fewer than 10 announcements in the list.

how about `int maxAnnouncements = std::min(10, (int)announcements.size());`?

Reply:
thx i forgot that.